storage/replica_command.go

```diff
@@ -1359,6 +1359,18 @@ func (r *Replica) splitTrigger(batch engine.Engine, split *roachpb.SplitTrigger)
 			// Our in-memory state has diverged from the on-disk state.
 			log.Fatalf("failed to update Store after split: %s", err)
 		}
+
+		// To avoid leaving the new range unavailable as it waits to elect
+		// its leader, one (and only one) of the nodes should start an
+		// election as soon as the split is processed. For simplicity, we
+		// choose the first node in the replica list. If this node is
+		// unavailable, the group will have to wait for an election
+		// timeout, just as with any other leader failure. (we could
+		// improve this by e.g. choosing the node that had the leader
+		// lease before the split and is therefore known to be up)
+		if r.store.StoreID() == split.NewDesc.Replicas[0].StoreID {
+			r.store.multiraft.Campaign(split.NewDesc.RangeID)
```

Comment:
As long as we drop and recreate the raft group, I don't see any way around this. It's not a huge problem in practice (because the worst case is just waiting for an election timeout), but the failure is frequent enough that the test will be flaky. I'm going to explore an alternate solution: instead of dropping the uninitialized group and recreating it as an initialized one, synthesize a snapshot (containing only metadata) to send to ourselves and initialize the group that way. If that works, it should make the test robust. If not, I'll have to fall back to either adding a 10ms delay (if that works consistently) or running the test multiple times and ensure that the election succeeds a majority of the time.